osu.Game/Overlays/ChatOverlay.cs

```diff
@@ -303,6 +303,18 @@ private void currentChannelChanged(ValueChangedEvent<Channel> e)
                 channelManager.MarkChannelAsRead(e.NewValue);
         }
 
+        /// <summary>
+        /// Highlights a certain message in the specified channel.
+        /// </summary>
+        /// <param name="message">The message to highlight.</param>
+        public void HighlightMessage(Message message)
+        {
+            if (channelManager.CurrentChannel.Value.Id != message.ChannelId)
+                channelManager.CurrentChannel.Value = channelManager.JoinedChannels.Single(c => c.Id == message.ChannelId);
```

Comment:
Not 100% sure about using `.Single` here rather than `.FirstOrDefault`. Most other usages of setting the current channel gracefully fallback rather than throwing. Seems like if a highlight fails we'd not want the game to crash.

Reply:
Now that I think about it, this could potentially fail if the notification comes from a channel that has now been left.

May wanna re-join the channel in that case... will see.